server/rust_ffi/rust_ffi.cc

```diff
@@ -0,0 +1,29 @@
+#include <iostream>
+#include <cstring>
+
+#include "rust_ffi.h"
+#include "../server.hh"
+
+using namespace seastar;
+
+extern "C" void store_from_rust(const char *key, const char *value) {
+    std::cout << "STORE$" << key << "$" << value << "$\n";
+
+    tcp_server::rust_data[std::string(key)] = std::string(value);
+}
+
+// Returns pointer to loaded value. If value is not found, returns NULL pointer.
+extern "C" char * load_from_rust(const char *key) {
+    std::cout << "LOAD$" << key << "$\n";
+
+    if (!tcp_server::rust_data.contains(std::string(key))) {
+        return NULL;
+    }
+
+    std::string value = tcp_server::rust_data[std::string(key)];
```

Comment:
Some nitpicks with regards to optimization:

- Both `.contains()` and `operator[]` perform a lookup and need to construct an `std::string`. We could get away with only one lookup and std::string construction by using `.find()`.
- As we are using C++20 I think `unordered_map` supports heterogenous lookup and you could use `std::string_view` for indexing. However, I'm not sure how easy it is (I couldn't find any snippets on the internet) so feel free to disregard this nitpick completely if this is non-trivial.